app/models/payload_template.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+class PayloadTemplate < Template
+  audited
+
+  include Authorizable
+  extend FriendlyId
+  friendly_id :name
+  include Parameterizable::ByIdName
+
+  class << self
+    # we have to override the base_class because polymorphic associations does not detect it correctly, more details at
+    # http://apidock.com/rails/ActiveRecord/Associations/ClassMethods/has_many#1010-Polymorphic-has-many-within-inherited-class-gotcha
+    def base_class
+      self
+    end
+  end
+  self.table_name = 'templates'
```

Comment:
Hmmm can't we simply make Template an abstract model class instead? Quick grep shows that we only create instances in core tests. This is hacky.